functions/src/utils.ts

```diff
@@ -34,38 +43,46 @@ export enum ShareLinksCollection {
  * @param documentId Shortened url for which to fetch data.
  * @returns Promise containing the data for the given short url.
  */
-export async function getUrlDocumentDataById(documentId: string) {
+export async function getUrlDocumentDataById(
+  documentId: string
+): Promise<DataOrError<ShareLinkRegisterParams>> {
   const db = admin.firestore();
   const querySnapshot = await db
     .collection(SHARE_LINK_FIRESTORE_COLLECTION)
     .doc(documentId)
     .get();
   if (!querySnapshot.exists) {
-    throw new Error(`Document with id ${documentId} doesn't exist`);
+    return {
+      data: undefined,
+      error: `Share link with id ${documentId} does not exist`,
+    };
   }
-  return querySnapshot.data() as ShareLinkRegisterParams;
+  const data = querySnapshot.data() as ShareLinkRegisterParams;
+  return { data: data, error: undefined };
 }
 
 /**
  * Creates a unique id for a document in a given collection.
  *
- * @param collection Firestore collection to check for uniqueness
- * @returns
+ * @param seed String to use as the seed for the unique id.
+ * @returns Eight digit unique id.
  */
-export async function createUniqueId(
-  collection: admin.firestore.CollectionReference
-): Promise<string> {
-  const urlHash = crypto.randomBytes(5).toString("hex");
-  const documentWithHash = await collection.doc(urlHash).get();
-  if (documentWithHash.exists) {
-    console.log("Hash collision. Generating new hash.");
-    return createUniqueId(collection);
-  } else {
-    console.log(`Hash generated: ${urlHash}`);
-    return urlHash;
-  }
+export function createUniqueId(
+  seed?: string
+): string {
+  const urlHash = seed
+    ? crypto.createHash("sha256").update(seed, "utf8").digest("hex").slice(0, 8)
```

Comment:
Not sure if it's safe to take a slice of a hash like this? It's possible that two different hashes will start with the same 8 digits, though probably unlikely?